livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamHomeActivity.kt

```diff
@@ -23,50 +24,43 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
-import io.getstream.chat.android.client.ChatClient
-import io.getstream.chat.android.client.api.models.QuerySort
-import io.getstream.chat.android.client.models.Filters
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
-import io.getstream.chat.android.compose.viewmodel.channel.ChannelListViewModel
-import io.getstream.chat.android.compose.viewmodel.channel.ChannelViewModelFactory
-import io.getstream.chat.android.offline.ChatDomain
 import io.getstream.livestream.compose.R
 import io.getstream.livestream.compose.darkColorPalette
 import io.getstream.livestream.compose.lightColorPalette
 
 class LiveStreamHomeActivity : ComponentActivity() {
-    private val factory by lazy {
-        ChannelViewModelFactory(
-            ChatClient.instance(), ChatDomain.instance(),
-            QuerySort.desc("last_updated"),
-            Filters.and(
-                Filters.eq("type", "livestream"),
-                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: ""))
-            )
-        )
-    }
-
-    private val listViewModel: ChannelListViewModel by viewModels { factory }
+    private lateinit var sharedPref: SharedPreferences
 
     @ExperimentalMaterialApi
     @ExperimentalFoundationApi
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        sharedPref =
+            getSharedPreferences(getString(R.string.name_shared_pref), Context.MODE_PRIVATE)
 
         setContent {
```

Comment:
Why isn't this a component outside of the `setContent()` function? E.g. `LiveStreamHomeContent`. It's hard to change this, it violates the open closed principle & SRP for the `onCreate()` function.

Reply:
>Why isn't this a component outside of the setContent() function? E.g. LiveStreamHomeContent.

What do you mean by outside `setContent` it is ouside , look at line 42 
confused with the point 

Why change ref to shared preferences ?

Reply:
I put this comment on the `setContent()` function not shared prefs.

As far as the rest of it goes - it's not outside of `setContent()`. We explicitly discussed why it's bad to have hardcoded components within `setContent()`, such as the Scaffold you have below.

- We can't prototype quickly by changing it to something like a `MyCustomUi()` function.
- We can't add the Preview annotation to the `onCreate` function because it's not a composable.
- If we want to test this component we can't just call it, we have to run the entire Activity.

We agreed that we'll pull this out (similar to what you did for the `RenderLiveStream` component. Not sure why this is still not a separate function.